### PR TITLE
docs(resource): fix for missing @param

### PR DESCRIPTION
**Pull request by thejustinwalsh**
I am unsure of the exact wording to use here, however the samples all imply that you require a path to use this function, and the documentation for each method is missing the @param calling out the required path.

Happy to update verbiage used if you see it not being fit or accurate.

### CODE REVIEW

File: engine/gamesys/src/gamesys/scripts/script_resource.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@ namespace dmGameSystem
  * [icon:attention] This function can only be called within [ref:go.property] function calls.
  *
  * @name resource.material
+ * @param path [type:string] a path string to the source version of the resource
```
Comment (JCash) [inline engine/gamesys/src/gamesys/scripts/script_resource.cpp:49]: Note that the path argument is optional. This should be clarified.

**Review by JCash — CHANGES_REQUESTED** [review]
Looks good, just need a small clarification.

**Comment by JCash** [comment]
Regarding the wording, perhaps "resource path string" makes it clearer that the path is something we use at build time.

**Review by britzl — APPROVED** [review]
Thanks! Looks good!

**Pull request merged**